lib/presentation/login_page/login_page.dart

```diff
@@ -23,22 +32,57 @@ class _LoginPageState extends State<LoginPage> {
   final formKey = GlobalKey<FormState>();
   final TextEditingController _pass = TextEditingController();
   final TextEditingController _confirmPass = TextEditingController();
-
-  final passwordValidator = MultiValidator([
-    RequiredValidator(errorText: 'emptyPassword'.tr),
-    MinLengthValidator(8, errorText: 'shortPassword'.tr),
-    PatternValidator(r'(?=.*?[#?!@$%^&*-])', errorText: 'specialPassword'.tr)
-  ]);
+  final TextEditingController _email = TextEditingController();
 
   @override
   Widget build(BuildContext context) {
-    Widget firstChild = LoginButton(
-      pointToOnPress: () {
-        if (formKey.currentState!.validate()) {
-          print('valid');
+    final db = FirebaseFirestore.instance;
+    Widget firstChild = BlocListener<SignInCubit, SignInState>(
+      listener: (context, state) {
+        if (state.signInStatus == SignInStatus.error) {
+          errorDialog(context, state.error);
+        } else if (state.signInStatus == SignInStatus.success) {
+          AutocompleteShowDialog.loginShowDialog(
+            context: context,
+            contentText: 'Czy chcesz schynchronizować bieżące miasta?',
+            child: TextButton(
+              child: Text('Tak'),
+              onPressed: () {
+                Authentication.updateDataWhenRegisterAndLogin();
+                //This could be strange that u see 2x pop, but it helps me with the bug
+                Navigator.of(context).pop();
+                Navigator.of(context).pop();
+              },
+            ),
+            secondChild: TextButton(
+              child: Text('Nie'),
+              onPressed: () {
+                Authentication.clearAndUpdate();
+                Navigator.of(context).pop();
+                Navigator.of(context).pop();
+              },
+            ),
+          );
         }
       },
+      child: LoginButton(
+        pointToOnPress: () {
+          if (formKey.currentState!.validate()) {
+            context.read<SignInCubit>().state.signInStatus ==
+                    SignInStatus.submitting
+                ? null
+                : context
+                    .read<SignInCubit>()
+                    .SignIn(email: _email.text, password: _pass.text)
+                    .then(
+                      (_) => context.read<UserCubit>().getProfile(
+                          uid: fbAuth.FirebaseAuth.instance.currentUser?.uid),
+                    );
```

Comment:
What if SignIn method failed?  I don't know that using .then is good here. Let's talk about this file after tomorrow's daily.